src/services/users.js

```diff
@@ -0,0 +1,186 @@
+      res.boom.badRequest();
+    }
+  }
+
+  async blockUser (req, res) {
+    try {
+      const statuses = await UserStatuses.find().exec();
+      const statusBlocked = statuses.find(({ name }) => name === 'Blocked');
+      const statusDeleted = statuses.find(({ name }) => name === 'Deleted');
+      const user = await User.findOne({ _id: req.params.id }).exec();
+      if (!user) return res.boom.notFound();
+      const currentStatus = user.status;
+      if (
+        statusBlocked._id.toString() !== currentStatus.toString() &&
+        statusDeleted._id.toString() !== currentStatus.toString()
+      ) {
+        const user = await User.findByIdAndUpdate(
+          req.params.id,
+          { status: statusBlocked._id },
+          {
+            new: true
+          }
+        ).exec();
+        return user ? res.json(true) : res.boom.notFound();
+      } else {
+        return res.boom.badRequest('user already blocked');
+      }
+    } catch (err) {
+      res.boom.badRequest();
+    }
+  }
+
+  async deleteUser (req, res) {
+    try {
+      const statusDeleted = await UserStatuses.findOne({
+        name: 'Deleted'
+      }).exec();
+      const user = await User.findOne({ _id: req.params.id }).exec();
+      if (!user) return res.boom.notFound();
+      const currentStatus = user.status;
+      if (statusDeleted._id.toString() !== currentStatus.toString()) {
+        const user = await User.findByIdAndUpdate(
+          req.params.id,
+          { status: statusDeleted._id },
+          {
+            new: true
+          }
+        ).exec();
+        return user ? res.json(true) : res.boom.notFound();
+      } else {
+        return res.boom.badRequest('user already deleted');
+      }
+    } catch (err) {
+      res.boom.badRequest();
+    }
+  }
+
+  async #createBaseUser (req, res, userRole) {
+    try {
+      const statuses = await UserStatuses.find().exec();
+      const statusActive = statuses.find(({ name }) => name === 'Active');
+      const statusDeleted = statuses.find(({ name }) => name === 'Deleted');
+
+      const users = await User.find({ email: req.body.email }).exec();
+      const activeUser = users.find(
+        (user) => user.status.toString() === statusActive._id.toString()
+      );
+      const deletedUser = users.find(
+        (user) => user.status.toString() === statusDeleted._id.toString()
+      );
+
+      if (!users.length || (deletedUser && !activeUser)) {
+        const newUser = new User(req.body);
+        newUser.status = statusActive._id;
+        const salt = await bcrypt.genSalt(10);
+        newUser.password = await bcrypt.hash(newUser.password, salt);
+        const roleUser = await UserRoles.findOne({ name: userRole }).exec();
+        newUser.role = roleUser._id;
+        await newUser.save();
+        return newUser ? res.json(true) : res.boom.notFound();
+      } else {
+        res.boom.badRequest('user already exists');
+      }
+    } catch (err) {
+      res.boom.badRequest();
```

Comment:
You need to throw here internal server error instead of badRequest. Fix this everywhere before merging